Project11/GameScene.swift

```diff
@@ -7,40 +7,174 @@
         makeBouncer(at: CGPoint(x: 1024, y: 0))
-
         
     }
     override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
         guard let touch = touches.first else { return }
         let location = touch.location(in: self)
+        let objects = nodes(at: location)
         
-        let ball = SKSpriteNode(imageNamed: "ballRed")
-        ball.physicsBody = SKPhysicsBody(circleOfRadius: ball.size.width / 2.0)
-        ball.physicsBody?.restitution = 0.4
-        ball.position = location
-        addChild(ball)
-    
+        if objects.contains(editLabel) {
+            editingMode.toggle()
+        } else {
+            if editingMode {
+                let size = CGSize(width: Int.random(in: 16...128), height: 16)
+                let box  = SKSpriteNode(color: UIColor(red: CGFloat.random(in: 0...1), green: CGFloat.random(in: 0...1), blue: CGFloat.random(in: 0...1), alpha: 1), size: size)
+                box.zRotation = CGFloat.random(in: 0...3)
+                box.position = location
+                
+                box.physicsBody = SKPhysicsBody(rectangleOf: box.size)
+                box.physicsBody?.isDynamic = false
+                box.name = "boxes"
+                addChild(box)
+                
+            } else {
+                var ballArray = ["ballRed", "ballBlue", "ballCyan", "ballGreen", "ballGrey", "ballPurple", "ballYellow"]
+                ballArray.shuffle()
+                
+                let ball = SKSpriteNode(imageNamed: ballArray[0])
+                ball.physicsBody = SKPhysicsBody(circleOfRadius: ball.size.width / 2.0)
+                ball.physicsBody?.restitution = 0.4
+                ball.physicsBody?.contactTestBitMask = ball.physicsBody?.collisionBitMask ?? 0
+                ball.position = location
+                ball.position.y = 720
+                ball.name = "ball"
+                if ballCounter == 0 {
+                    
+                } else {
+                    addChild(ball)
+                    ballCounter -= 1
+                }
+            }
+        }
     }
     
-    func makeBouncer(at position: CGPoint) {
+    private func makeBouncer(at position: CGPoint) {
         let bouncer = SKSpriteNode(imageNamed: "bouncer")
         bouncer.position = position
         bouncer.physicsBody = SKPhysicsBody(circleOfRadius: bouncer.size.width / 2)
         bouncer.physicsBody?.isDynamic = false
         addChild(bouncer)
     }
+    
+    private func makeSlot(at position: CGPoint, isGood: Bool) {
+        var slotBase: SKSpriteNode
+        var slotGlow: SKSpriteNode
+        
+        if isGood {
+            slotBase = SKSpriteNode(imageNamed: "slotBaseGood")
+            slotGlow = SKSpriteNode(imageNamed: "slotGlowGood")
+            slotBase.name = "good"
+        } else {
+            slotBase = SKSpriteNode(imageNamed: "slotBaseBad")
+            slotGlow = SKSpriteNode(imageNamed: "slotGlowBad")
+            slotBase.name = "bad"
+        }
```

Comment:
Small suggestion: this can be simplified to

```suggestion
        let slotBase: SKSpriteNode = isGood ? SKSpriteNode(imageNamed: "slotBaseGood") : SKSpriteNode(imageNamed: "slotBaseBad")
        slotBase.name = isGood ? "good" : "bad"
        let slotGlow: SKSpriteNode = isGood ? slotGlow = SKSpriteNode(imageNamed: "slotGlowGood") : SKSpriteNode(imageNamed: "slotGlowBad")
```